### DISCUSSION TITLE
CH01.08 테라폼 실행 관련 문의 입니다

### DISCUSSION THREAD

**Discussion by niceguykim** [tedilabs/fastcampus-devops#66] [Q&A] [answered]
서버 STOP /Create를 반복하게 될듯 하여 terraform  사용법을 익혀 보려고 하는데요
첨부와 같은 오류가 납니다 제가 terraform은 아예 처음 사용해 보는데 관련 레퍼런스가 별로 없고 주변에 아는 사람들이 없어 진행이 쉽지는 않네요
초반인데도 이렇게 삐걱거리는걸 보면 앞으로도 쉽지 않을것 같다는 생각이 드는데요 
terrafrom 사용과 원할한 수업 진행을위해서는 별도의 terraform 선수지식이 필요한 것인지.. 아니면 앞으로 강의 들으면서 대략적으로 이해할만한 정도의 사용인지도 궁금합니다
![image](https://user-images.githubusercontent.com/105776915/168988391-00fb4d08-a62e-4cae-b8d6-eb9d9e88c236.png)

**Comment by posquit0** [comment, marked as answer]
안녕하세요,
강의 수강에 불편을 드려 죄송합니다... 제가 기존에 진행하였던 패스트캠퍼스 <데브옵스 초격차> 강의 중 도커 컨테이너 기초 파트를, 패스트캠퍼스 측에서 <도커 & 쿠버네티스>의 초반 파트로 사용하면서 혼란을 드린 것 같습니다.

### 해결 방법

- <도커 & 쿠버네티스>를 수강중이시라면 테라폼을 굳이 익히지 않으셔도 될 것 같습니다. 남는 기계나 가상머신, AWS EC2 등에 리눅스 운영체제를 설치하여 실습 환경을 쉽게 마련하실 수 있습니다!
  - [가상머신 VMware에 Ubuntu 20.04 LTS 설치하기](https://tomcabin.tistory.com/3)
  - [AWS EC2에 (리눅스 기반)우분투 20.04LTS 인스턴스 생성하기 ](https://velog.io/@newon-seoul/AWS-EC2%EC%97%90-%EB%A6%AC%EB%88%85%EC%8A%A4-%EA%B8%B0%EB%B0%98%EC%9A%B0%EB%B6%84%ED%88%AC-20.04LTS-%EC%9D%B8%EC%8A%A4%ED%84%B4%EC%8A%A4-%EC%83%9D%EC%84%B1%ED%95%98%EA%B8%B0)
- 만약 테라폼을 이용해보고자 하신다면, 주어진 실습 코드에서 `config.yaml` 파일을 본인의 환경에 맞게 커스터마이징 해주셔야 합니다. 추가적으로 `aws-cli`를 설치하시고 본인의 AWS 계정으로 `aws-cli`를 이용할 수 있도록 사용자 인증 설정을 진행해주셔야 합니다.
